### PR TITLE
Add the document to backfill using tx_meta

### PR DESCRIPTION
**Pull request by amishas157**
<details>
  <summary>PR Checklist</summary>

### PR Structure

- [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
- [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
      otherwise).
- [ ] This PR's title starts with the jira ticket associated with the PR.

### Thoroughness

- [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
- [ ] I've updated the README with the added features, breaking changes, new instructions on how to use the repository. I updated the description of the fuction with the changes that were made.

### Release planning

- [ ] I've decided if this PR requires a new major/minor/patch version accordingly to
    [semver](https://semver.org/), and I've changed the name of the BRANCH to release/_ , feature/_ or patch/\* .
</details>

### What

This PR add instructions on backfill historical data using tx_meta

### Why

Airflow runs can be time and money consuming. To ease up the migrations, JS UDF can be convenient.

### Known limitations

To be figures: Cost of UDF (they should be cheap most likely)

### CODE REVIEW

**Comment by amishas157** [comment]
Steps outlined in the document added in this PR. I will add screenshots and links for confirming test results

## Library in GCS bucket

Screenshot 2024-07-31 at 2 41 17 PM

https://console.cloud.google.com/storage/browser/_details/stellar-test-js-udf/stellar-base.min.js;tab=live_object?project=crypto-stellar

## JS UDF

https://console.cloud.google.com/bigquery?ws=!1m7!1m6!12m5!1m3!1stest-hubble-319619!2sus-central1!3s834c1d13-e58e-40ff-90ec-a7435ec4fe42!2e1
Screenshot 2024-07-31 at 2 44 17 PM


## Results confirmations:
Screenshot 2024-07-31 at 2 45 52 PM


Sample check against horizon ✅ 

-  **transaction_hash**: f5f5b0aaf758896ef8c5b4807f41c77d15c11977eecf2b0e4769d777324a2d11
    **js_udf_result**: MCBD54KAHHA4AK4DOZWOSX5O5OZ4OI54N24QITDSFLPD7EG2WY2AMAAACYGGCDRL6UBUA
    **horizon_result**: MCBD54KAHHA4AK4DOZWOSX5O5OZ4OI54N24QITDSFLPD7EG2WY2AMAAACYGGCDRL6UBUA
Screenshot 2024-07-31 at 2 48 18 PM


---

**Next steps:**

- [ ] Discuss and get a review from the team
- [ ] Add js library to final bucket. Bucket to be decied.
- [ ] Add SQL snippet to update column from the results
- [ ] Run the query for historical data

File: backfill.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+LANGUAGE js
+OPTIONS (
+  library=["gs://stellar-test-js-udf/stellar-base.min.js"] -- path to js library in GCS
+)
+AS r"""
+    let txe = StellarBase.xdr.TransactionEnvelope.fromXDR(tx_meta, 'base64');
+    let tx = txe.feeBump();
+    let sourceAccount = StellarBase.encodeMuxedAccountToAddress(tx.tx().feeSource());
+    return sourceAccount
+
```
Comment (amishas157) [inline backfill.md:55]: Thanks @Shaptic for the helps here 🙌

File: backfill.md
Comment (chowbao) [inline backfill.md:1]: Very nice! 

Do you think we should make a new directory for this .md? It seems misplaced in the base directory. Maybe like a `notes` or `scripts` directory? Open to naming ideas

Comment (amishas157) [reply backfill.md:1]: `docs/`? I am also thinking to include `migrations` directory, where we manually track any udfs run we apply

Comment (chowbao) [reply backfill.md:1]: Yup I like that idea

**Review by chowbao — APPROVED** [review]
LGTM

File: backfill.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+yarn
+npx webpack --config config/webpack.config.browser.js
+```
+
+Above will create following files in `js-stellar-base/dist/` directory:
+
+- stellar-base.js
+- stellar-base.min.js
+
+Copy above files in a GCS bucket.
```
Comment (Shaptic) [inline backfill.md:25]: You only need the `min` file :+1:

File: backfill.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+Referred medium article: [Using NPM Library in Google BigQuery UDF](https://medium.com/analytics-vidhya/using-npm-library-in-google-bigquery-udf-8aef01b868f4)
+
+# Setting up JS UDF in Bigquery
+
+```
+git clone https://github.com/stellar/js-stellar-base.git
+cd js-stellar-base
+yarn
+npx webpack --config config/webpack.config.browser.js
```
Comment (Shaptic) [inline backfill.md:17]:
```suggestion
yarn
yarn build:prod
```
more idiomatic, I think

File: backfill.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+- stellar-base.js
+- stellar-base.min.js
+
+Copy above files in a GCS bucket.
+
+# Writing JS Function
+
+Following is example to extract FeeAccountMuxed:
+
+```
```
Comment (Shaptic) [inline backfill.md:31]: suggestion: add `javascript` tag here for syntax highlighting

File: backfill.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+CREATE TEMP FUNCTION getFeeBumpAccountIfExists(tx_meta STRING)
+RETURNS STRING
+LANGUAGE js
+OPTIONS (
+  library=["gs://stellar-test-js-udf/stellar-base.min.js"] -- path to js library in GCS
+)
+AS r"""
+    let txe = StellarBase.xdr.TransactionEnvelope.fromXDR(tx_meta, 'base64');
+    let tx = txe.feeBump();
+    let sourceAccount = StellarBase.encodeMuxedAccountToAddress(tx.tx().feeSource());
```
Comment (Shaptic) [inline backfill.md:54]: Nice, forgot I added this utility function lol

File: backfill.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    return sourceAccount
```
Comment (Shaptic) [inline backfill.md:55]:
```suggestion
    return StellarBase.encodeMuxedAccountToAddress(
        StellarBase.xdr.TransactionEnvelope.fromXDR(tx_meta, 'base64')
          .feeBump()
          .tx()
          .feeSource()
    );
```
:wink:

File: backfill.md [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+```
+CREATE TEMP FUNCTION getFeeBumpAccountIfExists(tx_meta STRING)
+RETURNS STRING
+LANGUAGE js
+OPTIONS (
+  library=["gs://stellar-test-js-udf/stellar-base.min.js"] -- path to js library in GCS
+)
+AS r"""
+    let txe = StellarBase.xdr.TransactionEnvelope.fromXDR(tx_meta, 'base64');
+    let tx = txe.feeBump();
```
Comment (Shaptic) [inline backfill.md:53]: Note that this will throw if it's not a fee bump

File: backfill.md (hunk 2 above)
Comment (amishas157) [reply backfill.md]: Yep. updating

File: backfill.md (hunk 3 above)
Comment (amishas157) [reply backfill.md]: oh nice

File: backfill.md (hunk 7 above)
Comment (amishas157) [reply backfill.md]: I am sending only fee bump tx to the function. For weird reasons, js udf was crying on try/catch blocks 🤷‍♀️

File: backfill.md (hunk 3 above)
Comment (amishas157) [reply backfill.md]: Addressed in https://github.com/stellar/stellar-etl/pull/273/commits/d5884a2f89388cb43db21a46ececa753eaee3a9e

File: backfill.md (hunk 4 above)
Comment (amishas157) [reply backfill.md]: Addressed in https://github.com/stellar/stellar-etl/pull/273/commits/d5884a2f89388cb43db21a46ececa753eaee3a9e

**Pull request merged**